### PR TITLE
chore: open transaction in view

### LINKED ISSUE #45: Open Transaction In View
[Sentry warning](https://sentry.io/organizations/hei-school/issues/2987827513/?project=6160544&query=is%3Aunresolved&statsPeriod=14d#threads) `spring.jpa.open-in-view is enabled by default. Therefore, database queries may be performed during view rendering. Explicitly configure spring.jpa.open-in-view to disable this warning`

Vlad is against it, but we don't mind: we choose simplicity over performance on this one.
https://vladmihalcea.com/the-open-session-in-view-anti-pattern/

### PR DESCRIPTION
**Pull request by hei-ryan**
Closes #45

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -23,3 +23,4 @@ spring.datasource.username=postgres
 spring.flyway.locations=classpath:/db/migration
 spring.jpa.hibernate.ddl-auto=none
 spring.jpa.show-sql=false
+spring.jpa.open-in-view=false
```
Comment (hei-teacher) [inline src/main/resources/application.properties]: no, put it to true, which is the default value that we currently use. it's just that we want to rid of the warning raised by spring boot. if you look at the tests, they fail if you set it to false

**Review by hei-teacher — CHANGES_REQUESTED** [review]

**Comment by hei-ryan** [comment]
Related to [](https://www.baeldung.com/spring-open-session-in-view#alternatives), LazyInitializationExceptions should be potentially prevent when dealing with lazy associations.

File: src/main/resources/application.properties (hunk 1 above)
Comment (hei-ryan) [reply src/main/resources/application.properties]: Yes.
[https://sentry.io/organizations/hei-school/issues/3044063138/?project=6160544&query=is%3Aunresolved+is%3Afor_review+assigned_or_suggested%3A%5Bme%2C+none%5D&sort=inbox&statsPeriod=14d](https://sentry.io/organizations/hei-school/issues/3044063138/?project=6160544&query=is%3Aunresolved+is%3Afor_review+assigned_or_suggested%3A%5Bme%2C+none%5D&sort=inbox&statsPeriod=14d)

Comment (hei-ryan) [reply src/main/resources/application.properties]: Related to https://www.baeldung.com/spring-open-session-in-view#alternatives, LazyInitializationExceptions should be potentially prevent when dealing with lazy associations.

**Review by hei-teacher — APPROVED** [review]

**Pull request merged**